File: Task1/circle.cpp
#include "circle.h"

// Ввод параметров круга из файла
void In(circle &c, ifstream &ifst) {
    string x, y, radius, colorNumber;
    ifst >> x >> y >> radius >> colorNumber;
    c.center.x = StringToInt(x);
    c.center.y = StringToInt(y);
    c.radius = StringToInt(radius);
    int color = StringToInt(colorNumber);
}

// Случайный ввод параметров круга
void InRnd(circle &c) {
    c.center.x = Random();
    c.center.y = Random();
    c.radius = Random();
}

// Вывод параметров круга в форматируемый поток
void Out(circle &c, ofstream &ofst) {
    ofst << "It is a Circle: x = " << c.center.x << ", y =" <<
    c.center.y << ", radius = " << c.radius << ". Area = " << Area(c);
}

// Вычисление площади круга
double Area(circle &c) {
    return 3.14 * pow(c.radius, 2);
}

File: Task3/shapes/circle.py
import math

from extensions.random_extension import get_random_int, get_random_point, get_random_color
from shapes.shape import Shape
from shapes.point import Point
from shapes.color import Color


# A circle class.
class Circle(Shape):
    __radius: int
    __center: Point

    # Circle init.
    def __init__(self, radius: int, center: Point, color: Color):
        Shape.__init__(self, color)
        self.__radius = radius
        self.__center = center

    # Returns a string with circle data.
    def to_string(self) -> str:
        return "It is a {color} circle: center: " \
               "{center}, radius: {radius}. Area = {area}". \
            format(color=self.color.name, center=self.__center.to_string(),
                   radius=self.__radius, area=self.area())

    # Returns an area.
    def area(self) -> float:
        return math.pi * self.__radius ** 2

    # Returns a random circle.
    @classmethod
    def get_random_circle(cls):
        return cls(get_random_int(), get_random_point(), get_random_color())
